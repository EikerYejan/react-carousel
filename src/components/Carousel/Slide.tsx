import styled from 'styled-components'

type Props = React.HTMLAttributes<HTMLDivElement> & {
  children: React.ReactNode
}

const Wrapper = styled.div`
  height: 100%;
  display: inline-flex;
`

const CarouselSlide = ({ children, ...restProps }: Props) => {
  return <Wrapper {...restProps}>{children}</Wrapper>
}

export default CarouselSlide
