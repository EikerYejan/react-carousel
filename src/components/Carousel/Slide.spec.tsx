import { render } from '@testing-library/react'
import Slide from './Slide'

describe('Slide', () => {
  it('renders correctly', () => {
    const { baseElement } = render(
      <Slide>
        <h1>TEST</h1>
      </Slide>,
    )

    expect(baseElement).toMatchSnapshot()
    expect(baseElement).toBeInTheDocument()
    expect(baseElement).toHaveTextContent('TEST')
  })
})
