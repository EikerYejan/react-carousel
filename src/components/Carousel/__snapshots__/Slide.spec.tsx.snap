// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide renders correctly 1`] = `
.c0 {
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<body>
  <div>
    <div
      class="c0"
    >
      <h1>
        TEST
      </h1>
    </div>
  </div>
</body>
`;
